core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/SixteenByNineFrameLayout.kt

```diff
@@ -16,11 +17,51 @@ open class SixteenByNineFrameLayout: FrameLayout {
     constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
     constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(context, attrs, defStyleAttr)
 
+    var mCustomTouchListener: OnTouchListener? =null
+
     override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
         if (layoutParams.height == ViewGroup.LayoutParams.WRAP_CONTENT) {
             val sixteenNineHeight = View.MeasureSpec.makeMeasureSpec(View.MeasureSpec.getSize(widthMeasureSpec) * 9 / 16, View.MeasureSpec.EXACTLY)
             super.onMeasure(widthMeasureSpec, sixteenNineHeight)
         } else
             super.onMeasure(widthMeasureSpec, heightMeasureSpec)
     }
+
+    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
+        /*
+            For every touch event passed to a ViewGroup (in this case SixteenByNineFrameLayout),
+            the dispatchTouchEvent() method will be called.
+
+            The dispatchTouchEvent() method first calls onInterceptTouchEvent() of SixteenByNineFrameLayout
+            to check if the touch event should be passed to child views or not.
+                If touch event should not be passed to children views,
+                    then onTouchEvent() will be called
+                 Else
+                    nothing happens and onInterceptTouchEvent() will return to dispatchTouchEvent(),
+                    which will continue to propagate the touch event to all children until one of
+                    them handles it
+
+
+            In this case, we want SixteenByNineFrameLayout to call onTouchEvent() regardless of whether
+            children handle the event or not
+
+            *reference: https://developer.android.com/training/gestures/viewgroup#intercept
+
+         */
+
+
+        // Even though click and long click events are being called, the touch listener
+        // doesn't seem to work. So we need to register and use our own touch listener
+       val isTouchEventHandled: Boolean? = mCustomTouchListener?.onTouch(rootView,ev)
+
+        if(isTouchEventHandled == null || !isTouchEventHandled)
+            onTouchEvent(ev)
+
+        return super.onInterceptTouchEvent(ev)
+    }
+
+    override fun setOnTouchListener(l: OnTouchListener?) {
+        mCustomTouchListener = l;
+        super.setOnTouchListener(l) // this doesn't seem to register the touch listener
```

Comment:
What does this comment means? Why are we calling it if it doesn't work?

Reply:
I assumed it might have an effect later on. But I saw how it was implemented. It does nothing, so i've removed it now.